pkg/cluster/deploystorage.go

```diff
@@ -109,12 +109,22 @@ func (m *manager) deployStorageTemplate(ctx context.Context) error {
 	clusterStorageAccountName := "cluster" + m.doc.OpenShiftCluster.Properties.StorageSuffix
 	azureRegion := strings.ToLower(m.doc.OpenShiftCluster.Location) // Used in k8s object names, so must pass DNS-1123 validation
 
+	imageRegistryStorageAccountName, err := m.imageRegistryStorageAccountName()
+	if err != nil {
+		return err
+	}
+
+	err = validateImageRegistryStorageAccountName(imageRegistryStorageAccountName)
+	if err != nil {
+		return err
+	}
```

Comment:
I think we only run `deployStorageTemplate` at installation time and I believe we will always have storage account name. It is the old clusters which we need to worry about.

Reply:
I had a quick review of all the places which today (in `master` as of 98fc429ecefc51d669cc72dc7cd3963231c18f1c) reference [`ImageRegistryStorageAccountName` field](https://github.com/Azure/ARO-RP/blob/70a3f19716075a23e7324fb42f78c7a30be3c09c/pkg/api/openshiftcluster.go#L135) and it looks like we only have issue in `populateRegistryStorageAccountName`.

Unless I miss something, in all other places we will have this field populates.

EDIT: Potentially `ensureGatewayCreate`, but it is a related, but separate issue.